cmake/check_cuda.cmake

```diff
@@ -56,7 +56,3 @@ else()
   list(REMOVE_ITEM generator_srcs ${generator_cuda_srcs})
   add_compile_definitions(USE_CUDA=0)
 endif()
-
-if(USE_CUDA AND NOT EXISTS "${ORT_LIB_DIR}/${ONNXRUNTIME_PROVIDERS_CUDA_LIB}")
-  message(FATAL_ERROR "Expected the ONNX Runtime providers cuda library to be found at ${ORT_LIB_DIR}/${ONNXRUNTIME_PROVIDERS_CUDA_LIB}. Actual: Not found.")
-endif()
```

Comment:
We should still have this, unless it has become redundant for some other reason. What would happen now if `USE_CUDA` flag is used and the directory is not present?